lib/views/frontend/spree/checkout/payment/_stripe.html.erb

```diff
@@ -1,71 +1,61 @@
 
-<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
-<script type="text/javascript">
-  Stripe.setPublishableKey("<%= payment_method.preferred_publishable_key %>");
-</script>
-
+<script type="text/javascript" src="https://js.stripe.com/v3/"></script>
 <script>
   Spree.stripePaymentMethod = $('#payment_method_' + <%= payment_method.id %>);
+  let stripe = Stripe("<%= payment_method.preferred_publishable_key %>");
+  let elements = stripe.elements();
+  let card = elements.create('card', {
+    style: {
+      base: {
+        lineHeight: "20px"
+      }
+    }
+  });
 
-  var mapCC, stripeResponseHandler;
 
-  mapCC = function(ccType) {
-    if (ccType === 'MasterCard') {
-      return 'mastercard';
-    } else if (ccType === 'Visa') {
-      return 'visa';
-    } else if (ccType === 'American Express') {
-      return 'amex';
-    } else if (ccType === 'Discover') {
-      return 'discover';
-    } else if (ccType === 'Diners Club') {
-      return 'dinersclub';
-    } else if (ccType === 'JCB') {
-      return 'jcb';
-    }
-  };
+  function addStripeHiddenInput(form, paymentMethodId, name, value) {
+    form.append($("<input type='hidden' name='payment_source[" + paymentMethodId + "][" + name + "]' value='" + value + "'/>"));
+  }
 
-  stripeResponseHandler = function(status, response) {
-    var paymentMethodId, token;
-    if (response.error) {
-      $('#stripeError').html(response.error.message);
-      return $('#stripeError').show();
-    } else {
-      Spree.stripePaymentMethod.find('#card_number, #card_expiry, #card_code').prop("disabled", true);
-      Spree.stripePaymentMethod.find(".ccType").prop("disabled", false);
-      Spree.stripePaymentMethod.find(".ccType").val(mapCC(response.card.brand));
-      token = response['id'];
-      paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][gateway_payment_profile_id]' value='" + token + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][last_digits]' value='" + response.card.last4 + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][month]' value='" + response.card.exp_month + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][year]' value='" + response.card.exp_year + "'/>");
-      return Spree.stripePaymentMethod.parents("form").get(0).submit();
-    }
+  function stripeTokenHandler(token) {
+    // Insert the token ID into the form so it gets submitted to the server
+    let form = $(Spree.stripePaymentMethod.parents("form").get(0));
+    let paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
+
```

Comment:
js supports nested functions.  You could put the `addStripeHiddenInput()` here and you wouldn't need to pass `form` and `paymentMethodId` into it as those would be in scope.

Another option to not having to pass these in every time is to define `addStripeHiddenInput` as a closure  This code is only used in this method, so it makes sense to isolate it.   I think this makes more sense.

```
let form = $(Spree.stripePaymentMethod.parents("form").get(0));
let paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
let addStripeHiddenInput = (name, value) => form.append($("<input type='hidden' name='payment_source[" + paymentMethodId + "][" + name + "]' value='" + value + "'/>"));
addStripeHiddenInput("gateway_payment_profile_id",  token.id);
```